### ISSUE TITLE
Window position with dynamic content is not correct + ng-if does not work in window

### ISSUE THREAD

**Issue by t8g** [angular-ui/angular-google-maps#1697]
I fight with 2 problems on window.

1. I can't manage to have correct positionned window with dynamic content.

![mapwindowwithstaticcontent](https://cloud.githubusercontent.com/assets/639759/12445253/87d8ba62-bf63-11e5-9c8c-546c99a07862.png)

![mapwindowwithdynamiccontent](https://cloud.githubusercontent.com/assets/639759/12445257/8dc30b3a-bf63-11e5-9be6-a8617a4245c7.png)

2. ng-if (nor ng-show) directive does'nt seems to work inside window

Here is a plunkr.
http://plnkr.co/edit/b2W55hb1GXiAIhP2w1ud?p=preview

An idea of what I'm doing wrong ?

Thanks

**Comment by david-meza** [comment]
TL;DR Here's a working Plnkr of what you were trying to do: http://plnkr.co/edit/K9hi3B?p=preview

Explanation:

1. When you pass the template as inline html, none of the angular directives seem to work. I've tried hacking my way around this and it may be possible to do it inline in your case, but the easiest solution by far is to just specify a templateUrl for your window.

2. The reason why the window wasn't correctly positioned is because you were setting the offset as `new google.maps.Size` You do not have access to this object by default. There are two ways to achieve the correct positioning:

    uiGmapGoogleMapApi.then(function (maps) {
      $scope.window = {
        options: {
          pixelOffset: new maps.Size(-28, -42)
        }
      };
    };

Where the `maps` is being passed in when the promise is resolved. This is the equivalent of `google.maps` object. Or:

    $scope.window = {
        options: {
          pixelOffset: { width: 0, height: -40 }
        },
      };

Where you just pass and object let the angular-google-maps library take care of running the function.

**Comment by t8g** [comment]
Well ... thanks, but the window doesn't appear to be dynamic at all.
It takes and keep the values of the first marker.

**Comment by david-meza** [comment]
My mistake, that should be fixed on the Plnker now. You can't replace the whole `$scope.window.templateParameter` object because it disconnects from the scope `$watch`. Change the attributes that are used by the info-window manually. The only issue I've come across doing this is that there's a flicker when the window resizes to the correct width and height, so if you figure out how to fix that let me know.

**Comment by t8g** [comment]
Ok thanks !

The flickering is another problem, but an acceptable one.

;)
